Composer/packages/lib/indexers/src/validations/index.ts

```diff
@@ -13,68 +12,59 @@ export const validateFuncs: { [type: string]: ValidateFunc[] } = {
   '.': [validateExpressions], //this will check all types
 };
 
-// check all fields
-function validateFields(
-  dialog: MicrosoftIDialog,
-  id: string,
-  schema: any,
-  lgFiles: LgFile[],
-  luFiles: LuFile[]
-): Diagnostic[] {
-  const diagnostics: Diagnostic[] = [];
-  /**
-   *
-   * @param path , jsonPath string
-   * @param value , current node value    *
-   * @return boolean, true to stop walk
-   * */
-  const visitor: VisitorFunc = (path: string, value: any): boolean => {
-    if (has(value, '$kind')) {
-      const allChecks = [...validateFuncs['.']];
-      const checkerFunc = validateFuncs[value.$kind];
-      if (checkerFunc) {
-        allChecks.splice(0, 0, ...checkerFunc);
-      }
-
-      allChecks.forEach((func) => {
-        const result = func(path, value, value.$kind, schema.definitions[value.$kind], lgFiles, luFiles);
-        if (result) {
-          diagnostics.splice(0, 0, ...result);
-        }
-      });
-    }
-    return false;
-  };
-  JsonWalk(id, dialog, visitor);
-  return diagnostics.map((e) => {
-    e.source = id;
-    return e;
-  });
-}
-
 export function validateDialog(
   dialog: DialogInfo,
   schema: SchemaDefinitions,
+  settings: DialogSetting,
   lgFiles: LgFile[],
   luFiles: LuFile[]
 ): Diagnostic[] {
   const { id, content } = dialog;
   try {
-    return validateFields(content, id, schema, lgFiles, luFiles);
+    const diagnostics: Diagnostic[] = [];
+    /**
+     *
+     * @param path , jsonPath string
+     * @param value , current node value    *
+     * @return boolean, true to stop walk
+     * */
+    const visitor: VisitorFunc = (path: string, value: any): boolean => {
+      if (has(value, '$kind')) {
+        const allChecks = [...validateFuncs['.']];
+        const checkerFunc = validateFuncs[value.$kind];
+        if (checkerFunc) {
+          allChecks.splice(0, 0, ...checkerFunc);
+        }
+
+        allChecks.forEach((func) => {
+          const result = func(path, value, value.$kind, schema.definitions[value.$kind], settings, lgFiles, luFiles);
+          if (result) {
+            diagnostics.splice(0, 0, ...result);
```

Comment:
Is there a reason we slice the result into diagnostics? Seems to warrant a comment if so and if not, this should just be a `push` instead.